[JIT] Include custom_class.h in torch/script.h

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32658 [JIT] Fix stateful lambda stuff and simplify code in custom C++ binding API
* #32604 [JIT] pickle serialization for custom bound classes
* **#32586 [JIT] Include custom_class.h in torch/script.h**
* #32555 [JIT] Serialize attributes and types in ClassType serialization
* #32553 [JIT] Support for registering C++ lambdas as methods on custom C++ class

Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

Commit:
[JIT] Include custom_class.h in torch/script.h

ghstack-source-id: c5e0362d0011eca27b5c1485219b8244fd952e03

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"




[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit a0170fdc:


**None of the build failures appear to be your fault**.

* **2/2** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=a0170fdc62128e9d676c90380dea8e0bdaac9bdf).

### :snowflake: 2 failures recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4333398) pytorch_linux_xenial_cuda10_1_cudnn7_py3_NO_AVX2_test (1/2)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=90417827) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=90417827)) :snowflake:
<details>
<summary>
<code>Jan 24 19:19:37 AssertionError: 10 not less than or equal to 1e-05 :</code>
</summary>

```
Jan 24 19:19:37 ---------------------------------------------------------------------- 
Jan 24 19:19:37 Traceback (most recent call last): 
Jan 24 19:19:37   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 174, in wrapper 
Jan 24 19:19:37     self._join_processes(fn) 
Jan 24 19:19:37   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 255, in _join_processes 
Jan 24 19:19:37     self._check_return_codes(elapsed_time) 
Jan 24 19:19:37   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 275, in _check_return_codes 
Jan 24 19:19:37     self.assertEqual(p.exitcode, first_process.exitcode) 
Jan 24 19:19:37   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 892, in assertEqual 
Jan 24 19:19:37     super(TestCase, self).assertLessEqual(abs(x - y), prec, message) 
Jan 24 19:19:37 AssertionError: 10 not less than or equal to 1e-05 :  
Jan 24 19:19:37  
Jan 24 19:19:37 ---------------------------------------------------------------------- 
Jan 24 19:19:37 Ran 117 tests in 151.965s 
Jan 24 19:19:37  
Jan 24 19:19:37 FAILED (failures=1, skipped=7) 
Jan 24 19:19:37  
Jan 24 19:19:37 Generating XML reports... 
Jan 24 19:19:37 Traceback (most recent call last): 
Jan 24 19:19:37   File "test/run_test.py", line 456, in <module> 
Jan 24 19:19:37     main() 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4333397) pytorch_linux_xenial_cuda10_1_cudnn7_py3_NO_AVX_NO_AVX2_test (2/2)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=90417896) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=90417896)) :snowflake:
<details>
<summary>
<code>Jan 24 19:19:42 AssertionError: 10 not less than or equal to 1e-05 :</code>
</summary>

```
Jan 24 19:19:42 ---------------------------------------------------------------------- 
Jan 24 19:19:42 Traceback (most recent call last): 
Jan 24 19:19:42   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 174, in wrapper 
Jan 24 19:19:42     self._join_processes(fn) 
Jan 24 19:19:42   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 255, in _join_processes 
Jan 24 19:19:42     self._check_return_codes(elapsed_time) 
Jan 24 19:19:42   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 275, in _check_return_codes 
Jan 24 19:19:42     self.assertEqual(p.exitcode, first_process.exitcode) 
Jan 24 19:19:42   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 892, in assertEqual 
Jan 24 19:19:42     super(TestCase, self).assertLessEqual(abs(x - y), prec, message) 
Jan 24 19:19:42 AssertionError: 10 not less than or equal to 1e-05 :  
Jan 24 19:19:42  
Jan 24 19:19:42 ---------------------------------------------------------------------- 
Jan 24 19:19:42 Ran 117 tests in 152.270s 
Jan 24 19:19:42  
Jan 24 19:19:42 FAILED (failures=1, skipped=7) 
Jan 24 19:19:42  
Jan 24 19:19:42 Generating XML reports... 
Jan 24 19:19:42 Traceback (most recent call last): 
Jan 24 19:19:42   File "test/run_test.py", line 456, in <module> 
Jan 24 19:19:42     main() 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]

Commit:
Update on "[JIT] Include custom_class.h in torch/script.h"


Differential Revision: [D19558716](https://our.internmc.facebook.com/intern/diff/D19558716)

[ghstack-poisoned]
